src/server/game/DataStores/DB2Structure.h

```diff
@@ -98,8 +98,8 @@ struct AreaTableEntry
     // Checks if zone activates pvp talents.
     bool ActivatesPvpTalents() const
     {
-        return (!IsSanctuary() && (Flags[0] & AREA_FLAG_ARENA) != 0)
-            || ID == 7618 || ID == 8320 || ID == 8304 || ID == 8151 || ID == 8078 || ID == 8051 || ID == 8324 || ID == 8168 || ID == 8328;
+        return (!IsSanctuary() && (Flags[0] & (AREA_FLAG_ARENA | AREA_FLAG_WINTERGRASP)) != 0)
+            || ID == 8320 || ID == 8304 || ID == 8078 || ID == 8051 || ID == 8324 || ID == 8328 || ID == 5095;
```

Comment:
For warden tower zones:
AREA_FLAG_UNK9 | AREA_FLAG_ALLOW_DUELS
+- 5099 areas have this flag | +- 4472 areas have this flag

For tol barad:
AREA_FLAG_OUTLAND2 | AREA_FLAG_ALLOW_DUELS

I think some flags have bad naming but I don't have the knowledge to research this :/